documentation/modules/operators/con-topic-operator-store.adoc

```diff
@@ -0,0 +1,115 @@
+The _topic store_ is based on the Kafka Streams key-value mechanism, which uses persistent storage with local and remote components.
+
+Topic metadata is stored and accessed locally in-memory.
+Updates from operations applied to the local in-memory topic store are persisted to a backup topic store on disk.
+The topic store is continually synchronized with updates from Kafka topics or Kubernetes `KafkaTopic` custom resources.
+Operations are handled rapidly with the topic store set up this way,
+but should the in-memory topic store crash it is automatically repopulated from the persistent storage.
+
+== Internal topic store topics
+
+Internal topics support the handling of topic metadata.
+
+`__strimzi_store_topic`:: Input topic for storing the topic metadata
+`__strimzi-topic-operator-kstreams-topic-store-changelog`:: Retains a log of compacted topic store values
+
+== Migrating topic metadata from ZooKeeper
+
+In previous releases of Strimzi, topic metadata was stored in ZooKeeper.
+The new process removes this requirement, bringing the metadata into the Kafka cluster, and under the control of the Topic Operator.
+
+When upgrading to Strimzi {ProductVersion}, the transition to Topic Operator control of the topic store is seamless.
+Metadata is found and migrated from ZooKeeper, and the old store is cleansed.
+
+== Downgrading to a Strimzi version that uses ZooKeeper to store topic metadata
+
+If you are reverting back to a version of Strimzi that uses ZooKeeper for the storage of topic metadata,
+you downgrade your Cluster Operator to the previous version,
+then downgrade Kafka brokers and client applications to the previous Kafka version as standard.
+
+Additionally, you must delete the topics created for the topic store using an `exec` command, specifying the bootstrap address of the Kafka cluster and the name of a broker pod.
+For example, where `my-cluster-kafka-0` is the name of the broker pod:
+
+[source,shell]
+----
+kubectl exec -ti my-cluster-kafka-0 -- ./bin/kafka-topics.sh --bootstrap-server localhost:9092 --topic __strimzi-topic-operator-kstreams-topic-store-changelog --delete && ./bin/kafka-topics.sh --bootstrap-server localhost:9092 --topic __strimzi_store_topic --delete
+----
+
+The Topic Operator will reconstruct the Zookeeper topic metadata from the state of the topics in Kafka.
+
+NOTE: When scaling a Kafka cluster, the topic store topics must also be deleted so that partitions can be reassigned.
+
+== Scaling and topics
+
+The recommended configuration for topics is to have a topic replication factor of 3, so that there are 2 in-sync replicas on other brokers.
+If a broker is unavailable, at least one other broker is in-sync.
+
+[source,shell,subs="+quotes,attributes"]
+----
+apiVersion: {KafkaTopicApiVersion}
+kind: KafkaTopic
+metadata:
+  name: my-topic
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  partitions: 1
```

Comment:
Is 1 partition just an example? Or is it a requirement? Can the TO work with multiple partitions @alesj?

Reply:
Hmmm, good question, since this is the topic controlled by Kafka Streams -- this is the input topic into our topology.
@tombentley ^^ how with the topics controlled by Kafka Streams?

Reply:
OK, afais, it goes like this:
* you (can) configure number of Streams threads

```   
public static final String NUM_STREAM_THREADS_CONFIG = "num.stream.threads";
private static final String NUM_STREAM_THREADS_DOC = "The number of threads to execute stream processing.";
```

and then those input topic partitions get distributed among all cluster threads / consumers.

Reply:
I can't think of a reason why it wouldn't work with >1 partition, but nor do I see that >1 would be needed.

Reply:
I can add another callout:

> <1> The number of partitions for the topic. Generally, 1 partition is sufficient.